vuln/SCHEMA.md

```diff
@@ -0,0 +1,42 @@
+# Schema for vulnerabilities
+
+| Key | Type | Description | Example |
+|-----|------|-------------|---------|
+|`id`|Integer > 0|Unique number identifying the record|349|
+|`created_at`|date-time|Record creation|"2017-05-19T22:45:59.16+00:00"|
+XXX do we need this, or is it just a mongo internal field?
+
+|`updated_at`|date-time|Record update|"2017-06-05T13:01:57.071+00:00"|
+XXX do we need this, or is it just a mongo internal field?
+
+|`title`|String|Short readable description|"Directory Traversal"|
+|`author`|String|Finder of vuln|"Liang Gong"|
+XXX what if reporter was not finder?
+
+|`module_name`|String|npm package name|"badjs-sourcemap-server"|
+|`publish_date`|date-time|when made public|"2017-06-05T13:01:57.07+00:00"|
+|`cves`|Array[String]|Associated CVE IDs|[]|
+XXX only 22 and 7 have multiple CVEs, might be simpler to enforce a CVE-per-record rule
+XXX for both, all the CVEs are reserved but still empty, whats up?
+XXX see http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2014-3743 for example
+
+|`vulnerable_versions`|semver-spec|vulnerable versions?|"<99.999.9999"|
+XXX What is exact syntax of semver-spec? Is it https://www.npmjs.com/package/semver#ranges ?
+
+|`patched_versions`|semver-spec|versions after fixed?|"<0.0.0"|
+XXX so vulnerable versions are `vulnerable_versions` MINUS `patched_versions`? And any other versions are assumed not vulnerable?
```

Comment:
i believe having the `vulnerable_versions` only is enough. (no need for patched_versions)